src/api/IBMi.ts

```diff
@@ -1240,17 +1259,21 @@ export default class IBMi {
    * @param {string} string
    * @returns {string} result
    */
-  sysNameInAmerican(string: string) {
-    const fromChars = this.variantChars.local;
-    const toChars = this.variantChars.american;
+  sysNameInAmerican(string: string, enabled: boolean = true) {
+    if (enabled) {
+      const fromChars = this.variantChars.local;
+      const toChars = this.variantChars.american;
 
-    let result = string;
+      let result = string;
 
-    for (let i = 0; i < fromChars.length; i++) {
-      result = result.replace(new RegExp(`[${fromChars[i]}]`, `g`), toChars[i]);
-    };
+      for (let i = 0; i < fromChars.length; i++) {
+        result = result.replace(new RegExp(`[${fromChars[i]}]`, `g`), toChars[i]);
+      };
+
+      return result;
+    }
 
-    return result
+    return string;
   }
```

Comment:
The logic behind the `enabled` parameter is unclear or would need to be redefined. Looking at all the references in the code, it's always this class' `qsysPosixPathsRequireTranslation` field that is used.

Here is a suggestion: make that second parameter optional, and when it's `true`, consider `qsysPosixPathsRequireTranslation` value. Then in the places where the method is called with the second parameter, pass `ifNeeded` as `true`:
```suggestion
  sysNameInAmerican(string: string, ifNeeded?: boolean) {
    if (!ifNeeded || this.qsysPosixPathsRequireTranslation) {
      const fromChars = this.variantChars.local;
      const toChars = this.variantChars.american;

      let result = string;

      for (let i = 0; i < fromChars.length; i++) {
        result = result.replace(new RegExp(`[${fromChars[i]}]`, `g`), toChars[i]);
      };

      return result;
    }

    return string;
  }
```